src/ape/managers/chain.py

```diff
@@ -937,7 +938,39 @@ def mine(
             self.pending_timestamp += deltatime
         self.provider.mine(num_blocks)
 
-    def set_balance(self, account: Union[BaseAddress, AddressType], amount: Union[int, str]):
+    def get_balance(
+        self, address: Union[BaseAddress, AddressType, str], block_id: Optional["BlockID"] = None
+    ) -> int:
+        """
+        Get the balance of the given address. If ``ape-ens`` is installed,
+        you can pass ENS names.
+
+        Args:
+            address (BaseAddress, AddressType | str): An address, ENS, or account/contract.
+            block_id (:class:`~ape.types.BlockID` | None): The block ID. Defaults to latest.
+
+        Returns:
+            int: The account balance.
+        """
+        if (isinstance(address, str) and not address.startswith("0x")) or not isinstance(
```

Comment:
we can make this simpler if this: https://github.com/ApeWorX/ape/pull/2521 merges